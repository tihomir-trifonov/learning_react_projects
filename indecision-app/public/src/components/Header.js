import React from "react";


// Header as stateless funktional component
const Header = (props) => {
    return (
        <div>
            <h1>{props.title}</h1>
            {props.subTitle && <h2>{props.subTitle}</h2>}
        </div>
    )
}

Header.defaultProps = {
    title: "Indecision App!!!"
}

export {Header as default}


// // Header as component with state
// class Header extends React.Component {
//     render() {
//         return (
//             <div>
//                 <h1>{this.props.title}</h1>
//                 <h2>{this.props.subTitle}</h2>
//             </div>
//         )
//     }
// };
