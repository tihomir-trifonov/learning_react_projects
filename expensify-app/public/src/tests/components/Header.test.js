import React from "react"
import Header from "../../components/Header"
import { shallow } from 'enzyme';
// import toJSON from "enzyme-to-json"

// import ReactShallowRendeder from "react-test-renderer/shallow"
// test("render Header correctly",()=>{
//     const renderer = new ReactShallowRendeder();
//     renderer.render(<Header/>)
//         expect(renderer.getRenderOutput()).toMatchSnapshot();
// })

test("test to render Header correctly",()=>{
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot()
    // expect(wrapper.find("h1").length).toBe(1)
    // expect(toJSON(wrapper)).toMatchSnapshot()

})