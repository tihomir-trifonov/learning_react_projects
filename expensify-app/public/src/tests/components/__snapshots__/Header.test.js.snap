// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit/1"
  >
    Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Get Help
  </NavLink>
</header>
`;
